### DISCUSSION TITLE
Automate the login process of the VNC client when using VNC protocol to communicate with docker-easyconnect container.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [docker-easyconnect/docker-easyconnect#37] [Ideas]
I try to find some script/API based method, say, using python, for automating the login process of VNC client but still haven't found any good solutions.

Do you have any good idea for this problem?

Regards,
HY

**Comment by Hagb** [comment]
Are "remember password" option and "auto login" option in EasyConnect available?

**Comment by hongyi-zhao** [comment]
I don't mean the connection process of EasyConnect itself, instead, I mean the login operation of VNC client. Obviously, you must first log into the remote VNC server before starting the EasyConnect client.

**Comment by Hagb** [comment]
Do you mean automatic login to VNC after starting the VNC server and before starting EasyConnect?

-----
> Obviously, you must first log into the remote VNC server before starting the EasyConnect client.

For login in EasyConnect, it is not needed to log into VNC before EasyConnect is started.
And after "auto login" option of EasyConnect is enabled, it will even be unnecessary to log in VNC, because after EasyConnect is started, the login of EasyConnect will be automatically done.

**Comment by Hagb** [comment]
https://linux.die.net/man/1/vncviewer
`-passwd` parameter of vncviewer may help.

**Comment by hongyi-zhao** [comment]
1. I really can't find any "auto login" option for EasyConnect on the login GUI of EasyConnect client included in your docker image published on docker hub. See the screenshot I observed:

![image](https://user-images.githubusercontent.com/11155854/99074482-c9a0bd00-25f2-11eb-87ab-70b7a542b216.png)

2. It seems that [RealVNC's vncviewer](https://www.realvnc.com/en/connect/download/viewer/linux/) is more convenient than that of [tigervnc](https://github.com/TigerVNC/tigervnc) to automatically preserve the login credentials.

**Comment by Hagb** [comment]
> I really can't find any "auto login" option for EasyConnect on the login GUI of EasyConnect client included in your docker image published on docker hub. See the screenshot I observed:

Maybe #7 can help.
And refer to https://github.com/Hagb/docker-easyconnect/pull/7#issuecomment-668639071 .

**Comment by hongyi-zhao** [comment]
> Maybe #7 can help.
> And refer to [#7 (comment)](https://github.com/Hagb/docker-easyconnect/pull/7#issuecomment-668639071) .

Wonderful and graceful ideas! Thanks a lot for letting me know this valuable discussion.

**Comment by Hagb** [comment]
(I tried to convert the issue to discussion.)
What about using the new [`cli`](https://github.com/Hagb/docker-easyconnect#%E7%BA%AF%E5%91%BD%E4%BB%A4%E8%A1%8C%E7%89%88) version with [`CLI_OPTS`](https://github.com/Hagb/docker-easyconnect#%E7%8E%AF%E5%A2%83%E5%8F%98%E9%87%8F) variable set?

**Comment by hongyi-zhao** [comment]
Tried with the latest git master cli version of docker-easyconnect but meet the same error:
```
werner@X10DAi:~$ touch ~/.easyconn
werner@X10DAi:~$ docker run --device /dev/net/tun --cap-add NET_ADMIN -ti -v $HOME/.easyconn:/root/.easyconn -p 127.0.0.1:1080:1080 -e EC_VER=7.6.3 hagb/docker-easyconnect:cli
vpn adress: 222.23.196.190:443                
username: <username>
password: <password>
Authenticate failed, reason: Invalid username or password!
Please reinput!
```

**Comment by Hagb** [comment]
I remember that your account seems disabled in #17

**Comment by hongyi-zhao** [comment]
The testing account belongs to the University where I have worked. So, I can't check if it has been really blocked/disabled.
